### ISSUE TITLE
parallel builds with make -jX sometimes broken

### ISSUE THREAD

**Issue by pierre-labastie** [libical/libical#423]
libical-3.0.8

When building at `-j5`, I get:
```
[ 98%] Generating ICalGLib-3.0.typelib
make[2]: Leaving directory '/sources/libical/libical-3.0.8/build'
error parsing file /sources/libical/libical-3.0.8/build/src/libical-glib/ICalGLib-3.0.gir: Error on line 1 char 1: Document was empty or contained only whitespace
make[2]: *** [src/libical-glib/CMakeFiles/gir-typelibs-ICalGLib_3_0_gir.dir/build.make:61: src/libical-glib/ICalGLib-3.0.typelib] Error 1
make[2]: Leaving directory '/sources/libical/libical-3.0.8/build'
make[1]: *** [CMakeFiles/Makefile2:680: src/libical-glib/CMakeFiles/gir-typelibs-ICalGLib_3_0_gir.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 98%] Built target gir-girs-ICalGLib_3_0_gir
```
When building at `-j4`, I just have:
```
[ 97%] Built target gir-girs-ICalGLib_3_0_gir
[ 98%] Generating ICalGLib-3.0.typelib
```
(no error). Grepping the source tree for "typelib", I found something weird: although everywhere in the *cmake/modules/GObjectIntrospectionMacros.cmake*, the directory in use seems to be `${CMAKE_CURRENT_BINARY_DIR}`, there is:
```
    add_custom_command(
      COMMAND ${INTROSPECTION_COMPILER}
              ${INTROSPECTION_COMPILER_ARGS}
              --includedir=.
              ${CMAKE_CURRENT_BINARY_DIR}/${gir}
              -o ${CMAKE_CURRENT_BINARY_DIR}/${_typelib}
      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${gir}
      OUTPUT ${_typelib}
      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    )
```
So why is `${CMAKE_CURRENT_SOURCE_DIR}` used at this point?
I've tried to chage this to `${CMAKE_CURRENT_BINARY_DIR}`, and the error at `-j5` disappeared (not tried anything else).

Note that I do not understand very well cmake macros, so I might be out of my depth here.

**Comment by winterz** [comment]
I have been unable to reproduce this. However, I think your proposed fix is correct.  
I'll give it a little bit of testing and commit if it works.

**Comment by pierre-labastie** [comment]
Thanks for looking at this. I must say I've never seen that again, although I rebuild systems quite often.

**Comment by winterz** [comment]
committed your suggested code fix in d1ac979884549926af495321c1cc3502b4a4551e
might help.  I don't think it can hurt.
